editoast/src/core/simulation.rs

```diff
@@ -476,6 +476,17 @@ impl AsCoreRequest<Json<SimulationResponse>> for SimulationRequest {
     }
 }
 
+impl SimulationResponse {
+    pub fn simulation_run_time(self) -> Result<u64, Box<Self>> {
+        match self {
+            SimulationResponse::Success { provisional, .. } => {
+                Ok(*provisional.times.last().expect("empty simulation result"))
```

Comment:
Are we 100% sure that Core doesn't send an empty list? If so I'd suggest making the message a bit more explicit

```suggestion
                Ok(*provisional.times.last().expect("core error: empty simulation result"))
```

If not, probably best to return an `Option<u64>` and let the caller deal with the situation.

Reply:
I updated the error message.
We need the result to forward it within this part of the code.
```rust
    let simulation_run_time = match virtual_train_sim_result.clone().simulation_run_time() {
        Ok(value) => value,
        Err(error) => {
            return Ok(Json(StdcmResponse::PreprocessingSimulationError {
                error: *error,
            }))
        }
    };
```

Reply:
> If not, probably best to return an Option<u64> and let the caller deal with the situation.

I meant to return an `Ok(Some(...))` instead, that was unclear, sorry